controllers/articlesController.js

```diff
@@ -93,4 +82,150 @@ export default class PostController {
       });
     }
   }
+
+  /**
+   * @description It helps the user to fetch all of the created articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async findAll(req, res) {
+    const { user: { id = null } = {} } = req;
+
+    try {
+      const all = await Article.findAll({
+        order: [["createdAt", "DESC"]],
+        include: [
+          {
+            model: User,
+            as: "author",
+            attributes: ["username", "bio", "image"],
+            include: [
+              { model: User, as: "followers", attributes: ["id", "username"] }
+            ]
+          },
+          { model: User, as: "likes", attributes: ["username"], through: {} },
+          { model: Tag, attributes: ["name"], as: "tagsList", through: {} }
+        ]
+      });
+      if (all.length) {
+        const articles = all.map(each => {
+          const {
+            author: { followers, username, bio, image }
+          } = each.get();
+          const { tagsList, likes, ...article } = each.get();
+          const following = !!_.dropWhile(followers, f => f.id !== id).length;
+          return {
+            ...article,
+            likes: likes.map(like => like.username),
+            author: { following, username, bio, image },
+            favorited: !!likes.length,
+            favoritesCount: likes.length,
+            tagsList: tagsList.map(t => t.get().name || null)
+          };
+        });
+        const articlesCount = articles.length;
+        return res
+          .status(OK)
+          .json({ message: "Successful", articles, articlesCount });
+      }
+      return res.status(NOT_FOUND).json({
+        message:
+          "No more articles found, you can also share your thoughts by creating an article"
+      });
+    } catch (error) {
+      return res
+        .status(INTERNAL_SERVER_ERROR)
+        .json({ message: "Can't get the articles, server error" });
+    }
+  }
+
+  /**
+   * @description It helps the user to update any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async update(req, res) {
+    try {
+      const { slug } = req.params;
+      const { user: { id: userId } = {}, body } = req;
+      const article = await Article.findOne({ where: { slug, userId } });
+      if (article) {
+        const allowed = _.pick(body, Object.keys(articleSchema));
+        const valid = await articleValidator(
+          allowed,
+          _.pick(articleSchema, Object.keys(allowed))
+        );
+        const { tagsList = [], ...rest } = valid;
+        if (tagsList.length) {
```

Comment:
@Musigwa Try to make it more meaningful by checking if tagsList.length is greater than 0 for better readability

Reply:
Hi @oesukam, thanks for the feedback. I'm not disagreeing with you but, I think this condition is more obvious because as you can see from the line above it, I'm destructuring the `tagsList` property and assign it to an empty array if not defined. This will make it able to have all the array prototype methods and properties in any situation. more than that, the length property is of type `number` and the minimum value it can have is `0` which is `false`. So, as the condition checks for `true` or `false`, I think from the explanation given, it proves how easy it is to understand the condition.

Reply:
@Musigwa It might be readable for you but not everyone. I tried to show other developers and it was not straight forward for them.
When talking about readability, `if (some_boolean == true)` is preferable than if `(some_boolean)` whereas `(some_boolean)` implies simplicity. They will always be some tradeoffs but what is important is to think about the team's interest

Reply:
Okay, let change it. thanks for clarifying that.